p2p/src/lib.rs

```diff
@@ -66,6 +67,27 @@ pub trait Receiver: Debug + Send + 'static {
     ) -> impl Future<Output = Result<Message<Self::PublicKey>, Self::Error>> + Send;
 }
 
+/// Interface for registering new peer sets as well as fetching an ordered list of connected peers, given a set id.
+pub trait PeerSetManager: Debug + Clone + Send + 'static {
+    /// Public key type used to identify peers.
+    type PublicKey: PublicKey;
+
+    /// The type for the peer set in registration.
+    type Peers;
+
+    /// Register a new ordered set of peers for a given ID.
+    fn register(&mut self, id: u64, peers: Self::Peers) -> impl Future<Output = ()> + Send;
```

Comment:
Only now realizing we use `register` for peer sets and for channels everywhere. I think we should delineate between the two.

Let's call this register `update` and keep the channel `register` as is.